vspace/src/main/webapp/WEB-INF/views/staff/modules/slides/slide.html

```diff
@@ -991,6 +1181,107 @@
+                            $('.error').append(alert);
+                            $(".error").delay(4000).slideUp(500, function() {
+                                $(".error").empty();
+                            });
+                            // Call the callback with null in case of error
+                            callback(null);
+                        }
+                    });
+                }
+                
+                function getCurrentSpace(blockId) {
+                    $.ajax({
+                        url: [[@{|/staff/spaces/search|}]],
+                        dataType: 'json',
+                        success: function(result) {
+                            let spaceName = null;
+                            for (let i = 0; i < result.length; i++) {
+                                const space = result[i];
+                                if (space.id === spaceId) {
+                                    spaceName = space.name;
+                                    break;
+                                }
+                            }
+                            callback(spaceName);
+                        },
+                        error: function(data) {
+                            var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>We are sorry but something went wrong. Please try again later.</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+                            $('.error').append(alert);
+                            $(".error").delay(4000).slideUp(500, function() {
+                                $(".error").empty();
+                            });
+                            // Call the callback with null in case of error
+                            callback(null);
+                        }
+                    });
+                }
+                
+	            $("#submitSpace").on("click", function(e) {
+		            $("#addSpaceAlert").hide();
+		            e.preventDefault();
+		            // ------------- creating space content blocks ------------
+		
+		            var formData = new FormData();
+		            var title = document.getElementById('spaceTitle').value;
+		            var selectedSpace = document.getElementById('spaceName');
+		            var token = [[${_csrf.token}]];
+		            if (selectedSpace.selectedIndex != -1 && selectedSpace.value != ""){
+
+			            var spaceId = selectedSpace.options[selectedSpace.selectedIndex].value;
+			            formData.append('title', title);
+			            formData.append('spaceId', spaceId);
+			            getSpaceById(spaceId, function(spaceName){
```

Comment:
for what does this need to be a callback?

Reply:
In fact, why does `getSpaceById` to be called here? The space block should only need the space and a title, which you have after selecting a space, don't you?